antenna/configlib.py

```diff
@@ -0,0 +1,323 @@
+                 raise_error=False):
+        """Returns a parsed value from the environment
+
+        :arg key: the key to look up
+        :arg namespace: the namespace for the key--different environments
+            use this differently
+        :arg default: the default value (if any)
+        :arg parser: the parser for converting this value to a Python object
+        :arg raise_error: True if you want a lack of value to raise a
+            ``ConfigurationError``
+
+        Examples::
+
+            # Use the special bool parser
+            DEBUG = config('DEBUG', default=True, parser=bool)
+
+            from antenna.config_util import ListOf
+            ALLOWED_HOSTS = config('ALLOWED_HOSTS', default='localhost',
+                                   parser=ListOf(str))
+
+        """
+        parser = get_parser(parser)
+
+        # Go through environments in reverse order
+        for env in self.envs:
+            val = env.get(key, namespace)
+            if val is not NO_VALUE:
+                return parser(val)
+
+        # Return the default if there is one
+        if default is not NO_VALUE:
+            return parser(default)
+
+        # No value specified and no default, so raise an error to the user
+        if raise_error:
+            raise ConfigurationError(
+                '%s (%s) requires a value of type %s' % (key, namespace, parser)
+            )
+
+        # Otherwise return None
+        return
+
+
+class ConfigManagerWrapper(object):
+    """Wraps the config manager so it's easier to set the config
+
+    This prevents the problem where Python modules load "config"
+    into their name space and then you have to do weird things to
+    fix that. Instead, you call ``config.set_config(CM)`` and pass
+    in a new ConfigManager and you're all set.
+
+    """
+    def __init__(self):
+        self.config = ConfigManager()
+
+    def set_config(self, config):
+        self.config = config
+
+    def __call__(self, *args, **kwargs):
+        return self.config.__call__(*args, **kwargs)
+
+
+# Use this when you're doing config things
+config = ConfigManagerWrapper()
+
+
+class ConfigOverride(object):
+    """Allows you to override config for tests
+
+    This can be used as a class decorator::
+
+        @config_override(FOO='bar', BAZ='bat')
+        class FooTestClass(object):
+            ...
+
+
+    This can be used as a function decorator::
+
+        @config_override(FOO='bar')
+        def test_foo():
+            ...
+
+
+    This can also be used as a context manager::
+
+        def test_foo():
+            with config_override(FOO='bar'):
+                ...
+
+    """
+    def __init__(self, **cfg):
+        self._cfg = cfg
+
+    def push_config(self):
+        _CONFIG_OVERRIDE.append(self._cfg)
+
+    def pop_config(self):
+        # Make sure it's not empty already because if it is, then we've done
+        # something horribly wrong.
+        assert _CONFIG_OVERRIDE
```

Comment:
Mmm... That's true now. It used to be that I had the other things in the environment stack so it was possible for `.pop_config()` to pop off the process environment or the ini environment.

I'll nix this because you're right in that it's not needed plus it's terribly unhelpful.